stylix/palette.nix

```diff
@@ -5,27 +5,88 @@
   options,
   ...
 }:
-
 let
   cfg = config.stylix;
-
 in
 {
+  imports = [
+    (lib.mkRemovedOptionModule
+      [
+        "stylix"
+        "polarity"
+      ]
+      "stylix: `stylix.polarity` has been removed in favor of `stylix.themeGeneration.polarity`, and its `either` value is no longer available"
+    )
+  ];
+
   options.stylix = {
-    polarity = lib.mkOption {
-      type = lib.types.enum [
-        "either"
-        "light"
-        "dark"
-      ];
-      default = "either";
-      description = ''
-        Use this option to force a light or dark theme.
+    themeGeneration = {
+      scheme = lib.mkOption {
+        type = lib.types.enum [
+          "content"
+          "expressive"
+          "fidelity"
+          "fruit-salad"
+          "monochrome"
+          "neutral"
+          "rainbow"
+          "tonal-spot"
+        ];
+        default = "tonal-spot";
+        description = "Use this option to select a color scheme type.";
+        apply = option: "scheme-${option}";
```

Comment:
I did not test this `apply` change. Hopefully, it works as expected.

Reply:
works